src/pages/settings/PasswordPage.js

```diff
@@ -139,15 +195,25 @@ class PasswordPage extends Component {
                                 autoCompleteType="password"
                                 textContentType="password"
                                 value={this.state.newPassword}
-                                onChangeText={newPassword => this.setState({newPassword})}
-                                onFocus={() => this.setState({isPasswordRequirementsVisible: true})}
-                                onBlur={() => this.onBlurNewPassword()}
+                                hasError={this.state.errors.newPassword || this.state.errors.newPasswordSameAsOld}
+                                errorText={!this.state.errors.newPassword && this.state.errors.newPasswordSameAsOld
+                                    ? this.getErrorText('newPasswordSameAsOld')
+                                    : this.getErrorText('newPassword')}
+                                onChangeText={text => this.clearErrorAndSetValue('newPassword', text, ['newPasswordSameAsOld'])}
                             />
-                            {this.state.isPasswordRequirementsVisible && (
-                                <Text style={[styles.textLabelSupporting, styles.mt1]}>
-                                    {this.props.translate('passwordPage.newPasswordPrompt')}
-                                </Text>
-                            )}
+                            {
+                              !this.state.errors.newPassword && !this.state.errors.newPasswordSameAsOld
```

Comment:
NAB, I would probably extract this to a variable for readability like `const shouldShowNewPasswordPrompt`